### PR TITLE
Add relation detection in analyze/text

### PR DESCRIPTION
**Pull request by toshev314**
### What's Changed?

- Added support for `relation_detection` in the analyze/text route, so coref can be enabled
- Updated locale field to be optional
### PR Checklist

- [x] Updated unit tests
- [x] Ran unit tests

### CODE REVIEW

File: src/privateai_client/components/request_objects.py [hunk 1]
```diff
@@ -970,6 +970,37 @@ def fromdict(cls, values: dict):
...
+    valid_coreference_resolutions = ["heuristics", "model_prediction", "combined"]
+
+    def __init__(self, coreference_resolution: str):
+        self._coreference_resolution = coreference_resolution
+
+    @property
+    def coreference_resolution(self):
+        return self._coreference_resolution
+
+    def _coreference_resolution_validator(self, var):
```
[the remark this answers is not in the document]
Comment (toshev314) [reply src/privateai_client/components/request_objects.py:992]: Thank you!

**Comment by toshev314** [comment]
@a-guiducci 
I also added some RelationDetection tests and changed the coref default to match the API (null)
